src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/AuthUser.kt

```diff
@@ -11,15 +11,19 @@ data class AuthUser(
   @NotNull val userName: String,
   var deleted: Boolean? = null,
 ) {
+  // Normally a user with a hmpps-auth Id represents a unique user. However, hmpps-auth has returned the same
+  // user with two different ids. To deal with this temporary issue, username is now being used for equality
+  // checking, e.g. don't email user if they are both the sender and assignee of a referral, as determined by user name.
+
   override fun hashCode(): Int {
-    return id.hashCode()
+    return userName.hashCode()
   }
 
   override fun equals(other: Any?): Boolean {
     if (other == null || other !is AuthUser) {
       return false
     }
 
-    return id == other.id
+    return userName == other.userName
```

Comment:
can we use `userName && authSource` ?

i think it's ever so slightly safer than just username (as long as it satifies all the constraints you have evaluated as part of this change).